### PR TITLE
Preprocess to allow lookup of intermediary members via number

### PR DESCRIPTION
**Pull request by i509VCB**
Implements the method and field part of #20 except for class name reordering

### CODE REVIEW

**Comment by i509VCB** [comment]
Wait it's all code style. 

🔫 Always has been

File: src/main/kotlin/net/fabricmc/bot/extensions/mappings/MappingsManager.kt [hunk 1]
```diff
@@ -91,7 +93,41 @@ class MappingsManager {
...
 
-        return getMappingsResults(mappings, query, ClassDef::getMethods)
+        return getMappingsResults(mappings, preProcessMethodQuery(query), ClassDef::getMethods)
+    }
+
+    /**
+     * Preprocesses a method mapping query, adapting the query value to make more ergonomic lookups possible.
+     */
+    private fun preProcessMethodQuery(query: String): String {
+        /*
```
Comment (gdude2002) [inline src/main/kotlin/net/fabricmc/bot/extensions/mappings/MappingsManager.kt]: This should be in the doc comment for the function — split it from the summary with two newlines

File: src/main/kotlin/net/fabricmc/bot/extensions/mappings/MappingsManager.kt [hunk 2]
```diff
@@ -91,7 +93,41 @@ class MappingsManager {
...
+         *  method_xxxx
+         *  xxxx -> method_xxxx
+         * Named:
+         *  blah
+        */
+
+        // Try intermediary path first.
+        // Since a method name per the Java Compiler cannot start with a number,
+        // we can assume the syntax is `xxxx` -> `method_xxxx`
+        if (query.isNotEmpty() && Character.isDigit(query[0])) {
```
Comment (gdude2002) [inline src/main/kotlin/net/fabricmc/bot/extensions/mappings/MappingsManager.kt]: Why check the first character and then check all the characters right after?

File: src/main/kotlin/net/fabricmc/bot/extensions/mappings/MappingsManager.kt [hunk 3]
```diff
@@ -104,7 +140,41 @@ class MappingsManager {
...
         val mappings = openMappings(minecraftVersion) ?: return null
 
-        return getMappingsResults(mappings, query, ClassDef::getFields)
+        return getMappingsResults(mappings, preProcessFieldQuery(query), ClassDef::getFields)
+    }
+
+    /**
+     * Preprocesses a field mapping query, adapting the query value to make more ergonomic lookups possible.
+     */
+    private fun preProcessFieldQuery(query: String): String {
```
Comment (gdude2002) [inline src/main/kotlin/net/fabricmc/bot/extensions/mappings/MappingsManager.kt:172]: See comments on the other function

**Review by gdude2002 — CHANGES_REQUESTED** [review]
A few comments. Also, is this not something you'd want for classes?

**Comment by i509VCB** [comment]
I believe all the issues have been addressed.

File: src/main/kotlin/net/fabricmc/bot/extensions/mappings/MappingsManager.kt [hunk 4]
```diff
@@ -76,11 +78,34 @@ class MappingsManager {
         val mappings = openMappings(minecraftVersion) ?: return null
 
         return mappings.classes
-                .filter { matches(it, query) }
+                .filter { matches(it, preProcessClassQuery(query)) }
```
Comment (gdude2002) [inline src/main/kotlin/net/fabricmc/bot/extensions/mappings/MappingsManager.kt]: Super pedantic, but this does mean that we're calling `preProcessClassQuery` for each filter iteration — might be best to store the pre-processed query in a variable before the return

**Review by gdude2002 — APPROVED** [review]

**Pull request merged**